src/main/java/se/sundsvall/document/api/DocumentResource.java

```diff
@@ -50,67 +52,69 @@
+
+	public DocumentResource(DocumentService documentService, ObjectMapper objectMapper) {
+		this.documentService = documentService;
+		this.objectMapper = objectMapper;
+	}
 
 	@PostMapping(consumes = { MULTIPART_FORM_DATA_VALUE }, produces = { ALL_VALUE, APPLICATION_PROBLEM_JSON_VALUE })
 	@Operation(summary = "Create document.")
 	@ApiResponse(responseCode = "201", headers = @Header(name = LOCATION, schema = @Schema(type = "string")), description = "Successful operation", useReturnTypeSchema = true)
 	public ResponseEntity<Void> create(
-		final UriComponentsBuilder uriComponentsBuilder,
-		@RequestPart("document") @Schema(description = "Document", implementation = Document.class) String documentString,
+		UriComponentsBuilder uriComponentsBuilder,
+		@RequestPart("document") @Schema(description = "Document", implementation = DocumentCreateRequest.class) String documentString,
 		@RequestPart(value = "documentFile") MultipartFile documentFile) throws JsonProcessingException {
 
-		final var document = mapper.readValue(documentString, Document.class); // If parameter isn't a String an exception (bad content type) will be thrown. Manual deserialization is necessary.
-		validate(document);
+		final var documentCreateRequest = objectMapper.readValue(documentString, DocumentCreateRequest.class); // If parameter isn't a String an exception (bad content type) will be thrown. Manual deserialization is necessary.
+		validate(documentCreateRequest);
 
-		// TODO: Call service layer.
-		final var id = document.getId();
+		final var registrationNumber = documentService.create(documentCreateRequest, documentFile).getRegistrationNumber();
 
-		return created(uriComponentsBuilder.path("/documents/{id}").buildAndExpand(id).toUri()).header(CONTENT_TYPE, ALL_VALUE).build();
+		return created(uriComponentsBuilder.path("/documents/{registrationNumber}").buildAndExpand(registrationNumber).toUri()).header(CONTENT_TYPE, ALL_VALUE).build();
 	}
 
 	@PatchMapping(path = "/{registrationNumber}", consumes = { MULTIPART_FORM_DATA_VALUE }, produces = { ALL_VALUE, APPLICATION_PROBLEM_JSON_VALUE })
 	@Operation(summary = "Update document.")
 	@ApiResponse(responseCode = "200", description = "Successful operation", useReturnTypeSchema = true)
 	@ApiResponse(responseCode = "404", description = "Not found", content = @Content(mediaType = APPLICATION_PROBLEM_JSON_VALUE, schema = @Schema(implementation = Problem.class)))
 	public ResponseEntity<Document> update(
-		@Parameter(name = "registrationNumber", description = "Document registration number", example = "2023-1337") @PathVariable("registrationNumber") String registrationNumber,
-		@RequestPart(value = "document", required = false) @Schema(description = "Document", implementation = Document.class) String documentString,
+		@Parameter(name = "registrationNumber", description = "Document registration number", example = "2023-2281-1337") @PathVariable("registrationNumber") String registrationNumber,
```

Comment:
Should the registration number be free-form as it currently is, or validated to conform to the given (example) format?

Reply:
Nah, skip it. Good reasoning